drivers/tps6274x/tps6274x.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2017 RWTH Aachen
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+
+#include "tps6274x.h"
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+static void print_err(char *, uint8_t *, uint8_t, uint8_t *);
+#ifdef TPS6274X_CONFIG
+const tps6274x_config_t converter_config = TPS6274X_CONFIG;
+#else
+#warning No config struct for module tps6274x available
+const tps6274x_config_t converter_config = {
+    .vsel1_pin = GPIO_UNDEF,
+    .vsel2_pin = GPIO_UNDEF,
+    .vsel3_pin = GPIO_UNDEF,
+    .vsel4_pin = GPIO_UNDEF,
+    .ctrl_pin = GPIO_UNDEF
+};
+#endif
+
+void print_err(char *err, uint8_t *vsel, uint8_t mask, uint8_t *vsel_not_set)
+{
+    if ((*vsel) & mask) {
+        printf("[tps6274x] Pin %s is not connected but is required for selected voltage level\n", err);
+        (*vsel_not_set) |= mask;
+    }
+
+}
+
+
+uint16_t tps6274x_init(uint16_t voltage)
+{
+    if (voltage < 1800) {
+        voltage = 1800;
+    }
+    else if (voltage > 3300) {
+        voltage = 3300;
+    }
+    uint8_t vsel = (voltage - 1800) / 100;
+    uint8_t vsel_not_set = 0;
+    if (converter_config.vsel1_pin != GPIO_UNDEF) {
+        gpio_init(converter_config.vsel1_pin, GPIO_OUT);
+        gpio_write(converter_config.vsel1_pin, vsel & 0b1);
+        DEBUG("vsel1: %u \n", vsel & 0b1);
+    }
+    else {
+
+        print_err("vsel1", &vsel, 0b1, &vsel_not_set);
+    }
+    if (converter_config.vsel2_pin != GPIO_UNDEF) {
+        gpio_init(converter_config.vsel2_pin, GPIO_OUT);
+        gpio_write(converter_config.vsel2_pin, vsel & 0b10);
+        DEBUG("vsel2: %u \n", vsel & 0b10);
+    }
+    else {
+        print_err("vsel2", &vsel, 0b10, &vsel_not_set);
+    }
+    if (converter_config.vsel3_pin != GPIO_UNDEF) {
+        gpio_init(converter_config.vsel3_pin, GPIO_OUT);
+        gpio_write(converter_config.vsel3_pin, vsel & 0b100);
+        DEBUG("vsel3: %u \n", vsel & 0b100);
+    }
+    else {
+        print_err("vsel3", &vsel, 0b100, &vsel_not_set);
+    }
+    if (converter_config.vsel4_pin != GPIO_UNDEF) {
+        gpio_init(converter_config.vsel4_pin, GPIO_OUT);
+        gpio_write(converter_config.vsel4_pin, vsel & 0b1000);
+        DEBUG("vsel4: %u \n", vsel & 0b1000);
+    }
+    else {
+        print_err("vsel4", &vsel, 0b1000, &vsel_not_set);
+    }
+    if (converter_config.ctrl_pin != GPIO_UNDEF) {
+        gpio_init(converter_config.ctrl_pin, GPIO_OUT);
+    }
+    else {
+        DEBUG("[TPS6274x] CTRL Pin not defined, no load activation possible");
+    }
+    return (vsel & (~vsel_not_set)) * 100 + 1800;
+}
+
+uint16_t tps6274x_switch_voltage(uint16_t voltage)
```

Comment:
seems to me that this function duplicates a lot of code of ```tps6274x_init()```. Maybe the latter can call the former?